model/jpa/src/main/resources/META-INF/jpa-changelog-20.0.0.xml

```diff
@@ -38,6 +39,18 @@
         </modifySql>
     </changeSet>
 
+    <changeSet author="keycloak" id="20.0.0-12964-supported-dbs-edb-migration">
+        <preConditions onSqlOutput="TEST" onFail="MARK_RAN">
+            <dbms type="postgresql"/>
+            <changeSetExecuted id="20.0.0-12964-supported-dbs" author="keycloak" changeLogFile="META-INF/jpa-changelog-20.0.0.xml"/>
+        </preConditions>
```

Comment:
As Keycloak sometimes skips to create indexes in the index has a lot of entries, it would be good to check if the index `IDX_GROUP_ATT_BY_NAME_VALUE` actually exists. I fell into that trap before :-(

Applies also for the other changesets.

Reply:
I added the precondition check for indexes,  it doesn't harm anything.

> As Keycloak sometimes skips to create indexes in the index has a lot of entries

Just out of curiosity, what do you mean? Did you mean "Keycloak sometimes skips to **drop** indexes if the index has a lot of entries"?